lib/ThreeScaleClient.php

```diff
@@ -95,7 +93,18 @@ public function getHost() {
    */
   public function authorize($appId, $appKey = null, $serviceId = null, $usage = null) {
     $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
-    $params = array('provider_key' => $this->getProviderKey(), 'app_id' => $appId);
+    $params = array('app_id' => $appId);
+
+    $providerKey = $this->getProviderKey();
+    $serviceToken = $this->getServiceToken();
+
+    if($providerKey) {
+      $params['provider_key'] = $providerKey;
+    }
+
+    if ($serviceToken) {
```

Comment:
I have added a [poc] commit. If approved, I will make changes to all other functions. Below would be the both workflows

**For service token flow**
1) The instance of client will be created using
```$client =  new ThreeScaleClient();```
2) The `authorize` function will be called using
```
$response = $client->authorize("the app id", "the app key", array("service_id" => "service id", "service_token" =>"service token"));
```

**For provider key flow**
1) The instance will be created using 
```$client =  new ThreeScaleClient("your provider key");```
2) The `authorize` function will be called using
```
$response = $client->authorize("the app id", "the app key", "service id");
```
In this way, we would be able to use same function without breaking changes. Let me know your feedback.